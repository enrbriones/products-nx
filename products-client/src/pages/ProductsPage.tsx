import React, { useEffect, useState } from 'react';

import ProductForm from '../components/ProductForm';
import ProductList from '../components/ProductList';
import {
  createProduct,
  deleteProduct,
  getProducts,
  Product,
} from '../services/productService';
import Modal from '../components/Modal';
import Button from '../components/Button';

const ProductsPage: React.FC = () => {
  const [products, setProducts] = useState<Product[]>([]);
  const [isModalOpen, setIsModalOpen] = useState(false);

  useEffect(() => {
    const fetchProducts = async () => {
      const fetchedProducts = await getProducts();
      setProducts(fetchedProducts);
      setIsModalOpen(false);
    };

    fetchProducts();
  }, []);

  const handleAddProduct = async (newProduct: Omit<Product, 'id'>) => {
    const createdProduct = await createProduct(newProduct);
    setProducts([...products, createdProduct]);
    setIsModalOpen(false);
  };

  const handleDeleteProduct = async (id: string) => {
    await deleteProduct(id);
    setProducts(products.filter((product) => product.id !== id));
  };

  const handleOpen = () => setIsModalOpen(true)
  const handleClose = () => setIsModalOpen(false)

  return (
    <div className="container mx-auto">
      <div className="text-center flex justify-between mt-6">
      <h1 className='text-2xl'>Productos</h1>
        <Button fn={handleOpen} label='Agregar producto' />
      </div>
      {isModalOpen && (
        <Modal close={handleClose} isOpen={isModalOpen}>
          <ProductForm onAdd={handleAddProduct} />
        </Modal>
      )}
      <ProductList products={products} onDelete={handleDeleteProduct} />
    </div>
  );
};

export default ProductsPage;
