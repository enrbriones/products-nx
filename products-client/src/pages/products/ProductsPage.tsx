import React, { useEffect, useState } from 'react';

import ProductForm from './components/ProductForm';
import ProductList from './components/ProductList';
import {
  createProduct,
  deleteProduct,
  getProducts,
} from './services/product.service';
import Modal from '../../components/Modal';
import Button from '../../components/Button';
import { Product } from './interfaces/product.interface';
import { useFetchProducts } from './hooks/useFetchProducts';

const ProductsPage: React.FC = () => {

  const {products, setProducts, isModalOpen, setIsModalOpen, reFetch} = useFetchProducts();

  const handleAddProduct = async (newProduct: Omit<Product, 'id'>) => {
    const createdProduct = await createProduct(newProduct);
    // setProducts([...products, createdProduct]);
    reFetch()
    setIsModalOpen(false);
  };

  const handleDeleteProduct = async (id: string) => {
    await deleteProduct(id);
    // setProducts(products.filter((product) => product.id !== id));
    reFetch()
  };

  const handleOpen = () => setIsModalOpen(true)
  const handleClose = () => setIsModalOpen(false)

  return (
    <div className="container mx-auto">
      <div className="text-center flex justify-between mt-6">
      <h1 className='text-2xl'>Productos</h1>
        <Button fn={handleOpen} label='Agregar producto' />
      </div>
      {isModalOpen && (
        <Modal close={handleClose} isOpen={isModalOpen}>
          <ProductForm onAdd={handleAddProduct} />
        </Modal>
      )}
      <ProductList products={products} onDelete={handleDeleteProduct} />
    </div>
  );
};

export default ProductsPage;
