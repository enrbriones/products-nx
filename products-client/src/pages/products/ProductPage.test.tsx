// src/components/ProductList.test.tsx
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ProductsPage from './ProductsPage';
// import '@testing-library/jest-dom/extend-expect'; // Para mÃ©todos adicionales como `toBeInTheDocument`

describe('ProductPage Component', () => {
  test('', () => {
    const {container}=render(<ProductsPage />);
    // screen.debug();
    expect(container).toMatchSnapshot()
  });
});
