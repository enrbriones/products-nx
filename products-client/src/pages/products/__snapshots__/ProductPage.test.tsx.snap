// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPage Component  1`] = `
<div>
  <div
    class="container mx-auto"
  >
    <div
      class="text-center flex justify-between mt-6"
    >
      <h1
        class="text-2xl"
      >
        Productos
      </h1>
      <button
        class="bg-blue-500 text-white px-3 py-1 rounded hover:bg-blue-600 mb-2 shadow-md"
      >
        Agregar producto
      </button>
    </div>
    <div
      class="p-4 mb-4 mt-8 text-sm text-blue-800 rounded-lg bg-blue-50 dark:bg-gray-800 dark:text-blue-400 text-center"
      role="alert"
    >
      <span
        class="font-medium"
      >
        No hay productos en la lista!
      </span>
    </div>
  </div>
</div>
`;
